### ISSUE TITLE
Custom.scss theming best practice. npm build.

### ISSUE THREAD

**Issue by swfeathers** [twbs/bootstrap#27191]
Through 4.0.0 alpha and beta I was able to build my own themed bootstrap.css using grunt or npm quite easily by following the relevant docs.

My custom.scss file was in the same folder as the rest of the bootstrap .scss files and was incorporated into the main css when I built, whether using grunt or npm.

I've just tried to do the same in 4.1.3 and it's been a bit of a mess.

The 'Importing' section of the 'Theme Bootstrap' doc seems to list the wrong path for the @import:

`@import "../node_modules/bootstrap/scss/bootstrap";`

Whether downloading the bootstrap source files manually or installing via npm, the "node-modules" folder is built at the root alongside the main 'scss', 'js', 'dist' folders, not as a parent as listed here. I'm sure you can change this, but I'm just saying for most people the docs are misleading.

Secondly, in ".stylelintrc" the line:

`scss/dollar-variable-default": [true, { "ignore": "local" }],`

causes build failures with any "custom.scss" file where the !default flags have been removed. The docs however say that removing the !default flags is the correct thing to do.

Basically, it would be good to get the theming page updated to include a best practice approach for basic customisation and build that works when you follow the instructions. Something like:

- Install Bootstrap via npm or download source files manually.
- Create a "custom.scss" file in a certain directory.
- Add these @import rules.
- Copy anything you want to edit from "_variables.scss".
- Add your own styles as required.
- Build by running npm et voila.

I've read about making custom package.json files as a workaround, which is fine too, but i twould be good to get a guide of the best and correct way to do this.

Thanks.

**Comment by XhmikosR** [comment]
You are not supposed to use our .stylelintrc in your project. At least not unmodified. The same goes for any meta files.

The current theming page has all the info https://getbootstrap.com/docs/4.1/getting-started/theming/

So, you were just doing it wrong all this time.

**Comment by swfeathers** [comment]
What I'm saying is that I used to follow the instructions from [here](https://v4-alpha.getbootstrap.com/getting-started/options/) and [here](https://v4-alpha.getbootstrap.com/getting-started/build-tools/) and it all made sense and worked correctly.

Perhaps most importantly, that was when bootstrap shipped with a `_custom.scss` file ready to be edited which made things very clear and straightforward.

I'm not alone in my frustration and reading [this thread](https://github.com/twbs/bootstrap/issues/25550) shows some of the confusion. [This](https://github.com/twbs/bootstrap/issues/25550#issuecomment-363645878) reply on the thread from mdo shows that current theming documentation is not clear because he's essentially adding to it on the fly.

If there is a different way to theme now, then that's fine, however I don't believe the current docs explain it well enough.

If we momentarily ignore building bootstrap yourself using npm, then is the documentation currently saying that I can just setup my own `custom.scss` file as instructed and use a sass compiler to create the main `.css` file for my website?

I'm sure getting this sorted is not on the priority list, but just try going to the [theming page](https://getbootstrap.com/docs/4.1/getting-started/theming/) and following the instructions word for word, pretending to be fairly new to using package managers and scss.

I'm almost certain that following your instructions you'd never end up with the file structure you list:
```
your-project/
├── scss
│   └── custom.scss
└── node_modules/
    └── bootstrap
        ├── js
        └── scss
```
I.e. `/bootstrap/` would not be a child within the `/node_modules/` folder.

Anyway, I'm just trying to help because I don't think the docs are as clear as they were beofre and I'm only telling you the real-world issues I've come across trying to following the new instructions. There are many people who may be trying to following these instructions and then simply giving up, which can't be great for bootstrap as a whole.

**Comment by XhmikosR** [comment]
You are more than welcome to provide a PR clearing up the theming page. I know it's something most people find confusing, I even had the issue when I jumped on v4 for my own projects.

That being said, currently, the theming page is the starting point for using Bootstrap in your project.
